src/control/cmd/agent/main.go

```diff
@@ -57,17 +57,69 @@ type cliOptions struct {
 	Insecure   bool       `short:"i" long:"insecure" description:"have agent attempt to connect without certificates"`
 	RuntimeDir string     `short:"s" long:"runtime_dir" description:"Path to agent communications socket"`
 	LogFile    string     `short:"l" long:"logfile" description:"Full path and filename for daos agent log file"`
+	NetworkDbg bool       `short:"n" long:"netdebug" description:"Enable extended network device scan debug output"`
 	Version    versionCmd `command:"version" description:"Print daos_agent version"`
+	NetScan    netScanCmd `command:"scan" description:"Perform local fabric scan"`
 }
 
 type versionCmd struct{}
+type netScanCmd struct {
+	Debug          bool   `short:"d" long:"debug" description:"Enable debug output"`
+	FabricProvider string `short:"p" long:"provider" description:"Filter device list to those that support the given OFI provider (default is all providers)"`
+}
 
 func (cmd *versionCmd) Execute(_ []string) error {
 	fmt.Printf("daos_agent version %s\n", build.DaosVersion)
 	os.Exit(0)
 	return nil
 }
 
+func (cmd *netScanCmd) Execute(args []string) error {
+	var provider string
+	defer os.Exit(0)
+	log := logging.NewCommandLineLogger()
+
+	if cmd.Debug {
+		log.WithLogLevel(logging.LogLevelDebug)
+		netdetect.SetLogger(log)
+	}
+
+	numaAware, err := netdetect.NumaAware()
+	if err != nil {
+		exitWithError(log, err)
+		return nil
+	}
+
+	if !numaAware {
+		fmt.Printf("This system is not NUMA aware")
+	}
+
+	switch {
+	case len(cmd.FabricProvider) > 0:
+		provider = cmd.FabricProvider
+		fmt.Printf("Scanning fabric for provider: %s\n", provider)
+	default:
+		fmt.Printf("Scanning fabric for all providers\n")
+	}
+
+	results, err := netdetect.ScanFabric(provider)
+	if err != nil {
+		exitWithError(log, err)
+		return nil
+	}
+
+	if provider == "" {
+		provider = "All"
+	}
+	fmt.Printf("\nFabric scan found %d devices matching the provider spec: %s\n\n", len(results), provider)
+
```

Comment:
It would be nice to have a --json option as well, to make programmatic usage of the output easier.

Reply:
I've added a first pass at enabling the JSON output.  I don't know if it should be formatted differently, or if the basic output is suitable.